lms/djangoapps/courseware/courses.py

```diff
@@ -519,6 +519,40 @@ def get_course_assignment_date_blocks(course, user, request, num_return=None,
     return date_blocks
 
 
+@request_cached()
+def get_course_blocks_completion_summary(course_key, user):
+    """
+    Returns an object with the number of complete units, incomplete units, and units that contain gated content
+    for the given course.
+
+    The object contains fields: complete_count, incomplete_count, locked_count
+    """
+    if not user.id:
+        return []
+    store = modulestore()
+    course_usage_key = store.make_course_usage_key(course_key)
+    block_data = get_course_blocks(user, course_usage_key, allow_start_dates_in_future=True, include_completion=True)
+
+    complete_count, incomplete_count, locked_count = 0, 0, 0
+    for section_key in block_data.get_children(course_usage_key):  # lint-amnesty, pylint: disable=too-many-nested-blocks
```

Comment:
I don't think you want the `lint-amnesty` part of this. The `lint-amnesty` was edX's way of saying, all of these lint errors are allowed in for historical purposes even though they are not correct just so we don't need to keep focusing on them when new PRs are made. I don't believe this would be included in that amnesty then because it is new.

That being said, ideally, we would want to remove the need for the disable, but I'm not entirely sure how in this case. You need to get down to the units (or rather leaves) and I think you need to do something like this where you keep going down children. It would be needlessly complicated to do some recursion or something just to avoid this pylint error. hmm. I guess I'd say to leave it.